mender-update/standalone/standalone.cpp

```diff
@@ -467,12 +534,48 @@ ResultAndError DoCommit(
 	context::MenderContext &main_context,
 	StateData &data,
 	update_module::UpdateModule &update_module) {
+	auto default_paths {main_context.GetConfig().paths};
+	events::EventLoop loop;
+	auto script_runner {executor::ScriptRunner(
+		loop,
+		chrono::seconds {main_context.GetConfig().state_script_timeout_seconds},
+		default_paths.GetArtScriptsPath(),
+		default_paths.GetRootfsScriptsPath())};
+	// Commit Enter
+	{
+		auto err =
+			script_runner.RunScripts(executor::State::ArtifactCommit, executor::Action::Enter);
+		if (err != error::NoError) {
+			log::Error("Commit Enter State Script error: " + err.String());
+			// Commit Error
+			{
+				auto commit_error = script_runner.RunScripts(
+					executor::State::ArtifactCommit, executor::Action::Error);
+				if (commit_error != error::NoError) {
+					log::Error("Commit Error State Script error: " + commit_error.String());
+				}
+			}
+			return InstallationFailureHandler(main_context, data, update_module);
+		}
+	}
+
 	auto err = update_module.ArtifactCommit();
 	if (err != error::NoError) {
 		log::Error("Commit failed: " + err.String());
 		return InstallationFailureHandler(main_context, data, update_module);
 	}
 
+	// Commit Leave
+	{
+		auto err =
+			script_runner.RunScripts(executor::State::ArtifactCommit, executor::Action::Leave);
+		if (err != error::NoError) {
+			auto leave_err =
+				script_runner.RunScripts(executor::State::ArtifactCommit, executor::Action::Error);
+			log::Error("Error during Commit Leave script: " + err.FollowedBy(leave_err).String());
```

Comment:
The result and error still needs to be kept here. You might want to move this block below the `result` and `return_err` declarations below, and then start doing the assignments like the handlers below that do.